src/Core/OrganizationFeatures/OrganizationGroups/DeleteGroupCommand.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using SendGrid.Helpers.Errors.Model;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationGroups;
+
+public class DeleteGroupCommand : IDeleteGroupCommand
+{
+    private readonly IGroupRepository _groupRepository;
+    private readonly IEventService _eventService;
+
+    public DeleteGroupCommand(IGroupRepository groupRepository, IEventService eventService)
+    {
+        _groupRepository = groupRepository;
+        _eventService = eventService;
+    }
+
+    public async Task DeleteAsync(Group group)
+    {
+        await _groupRepository.DeleteAsync(group);
+        await _eventService.LogGroupEventAsync(group, Enums.EventType.Group_Deleted);
+    }
+
+    public async Task<ICollection<Group>> DeleteManyAsync(Guid orgId, IEnumerable<Guid> groupIds)
+    {
+        var ids = groupIds as Guid[] ?? groupIds.ToArray();
+        var groupsToDelete = await _groupRepository.GetManyByManyIds(ids);
+        var filteredGroups = groupsToDelete.Where(g => g.OrganizationId == orgId).ToList();
```

Comment:
Thoughts on moving this logic out of the command so it only needs to take IDs and perform the action/return the result? Maybe into a helper or even in the controller.

cc: @MGibson1 Does this suggestion make sense per our conversation?

Reply:
My initial feeling is that this logic belongs in the command as it should likely be applied every time multiple groups are deleted (always ensuring the deleted groups belong to the expected organization). Otherwise, it will just be repeated in each calling controller/service which provides more surface area for mistakes, imo.

With that said, I'm open to hearing more on why you think it should be moved out of the command, I could be missing some context related to how we want CQRS commands to work.

Reply:
I'd like to see grabbing of organization and groups outside of the command, then a guard that throws if a group doesn't belong to the given org is reasonable.

However, even more reasonable would be not requiring an org at all. The controller should validate permissions to delete any groups it gives in, then at the command level it shouldn't care at all about the orgs.

It seems the way the sproc is written right now limits deleteMany to a single organization. I don't really understand why In a repository sense, there's no reason to restrict deletes of a group to a single org at a time. this is a business logic concern at the controller level (if even there, as long as you have permission, I don't really care if you delete for a single org or many)